sdk/storage/azure-storage-blob/samples/directory_interface.py

```diff
@@ -0,0 +1,303 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+'''
+FILE: directory_interface.py
+DESCRIPTION:
+    This example shows how to perform common filesystem-like operations on a
+    container. This includes uploading and downloading files to and from the
+    container with an optional prefix, listing files in the container both at
+    a single level and recursively, and deleting files in the container either
+    individually or recursively.
+
+    Fill in the required constant values and unzip directory_interface_data.zip
+    to run the samples below.
+'''
+
+import os
+from azure.storage.blob import BlobServiceClient
+
+# TODO: Fill in your connection string
+CONNECTION_STRING = ''
+CONTAINER_NAME = ''
```

Comment:
I'd recommend replacing this with code that pulls from environment variables:
```Python
import os
import sys

try:
  CONNECTION_STRING = os.environ['AZURE_STORAGE_CONNECTION_STRING']
except KeyError:
  print('AZURE_STORAGE_CONNECTION_STRING must be set.')
  sys.exit(1)
```

This way we don't have to worry about accidentally committing a credential and the script can theoretically be run "as-is".

Reply:
Container name could simply be an argument to the script.

Reply:
Good thought, it's addressed :)